src/ruby/lib/grpc/errors.rb

```diff
@@ -42,12 +42,12 @@ def initialize(code, details = 'unknown cause', metadata = {})
       @metadata = metadata
     end
 
-    # Converts the exception to a {Struct::Status} for use in the networking
+    # Converts the exception to a {GRPC::Core::Status} for use in the networking
     # wrapper layer.
     #
-    # @return [Struct::Status] with the same code and details
+    # @return [GRPC::Core::Status] with the same code and details
     def to_status
-      Struct::Status.new(code, details, metadata)
+      GRPC::Core::Status.new(code, details, metadata)
```

Comment:
I don't think changing the `Status` module will work for backwards compatibility reasons. Note that while `NewServerRpc` and `BatchResult` are relatively internal objects, `Struct::Status` is a part of the public surface API.

Can we solve the problem in https://github.com/grpc/grpc/pull/19893#issuecomment-521006344 by just moving the definition of `Status` out of the C extension?

Reply:
What about moving the definition and then aliasing the old name to the new name?

Reply:
Though this would work, it's a slight complication - what's the motivation?

Reply:
The motivation is only that it looks like the intention would have been to locate these classes in the `GRPC` namespace instead of using the default location under `Struct`.

In Ruby I would expect an anonymous struct to be assigned to a constant:

```ruby
GRPC::Core::Status = Struct.new(:code, :details, :metadata)
```

Instead of creating a named struct that uses the default location:

```ruby
Struct.new('Status', :code, :details, :metadata) # Struct::Status
```

Reply:
I am happy to put this on pause and revisit it again once #19939 has been considered.